### PR TITLE
Add Neo4j notebook

### PR DESCRIPTION
**Pull request by benofben**
This is the fourth!!!! PR for the Neo4j and Vertex notebook @htappen and I have been working on.  We've been having CLA issues getting it submitted.

If you are opening a PR for `Community Notebooks` under the [notebooks/community](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/community) folder:
- [ X] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/docs/CODEOWNERS) file under the `# Community Notebooks` section, pointing to the author or the author's team.
- [ X] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/docs/contributing.md#code-quality-checks).

### CODE REVIEW

**Comment by benofben** [comment]
I see it's now blocking on "ERROR: notebooks/community/neo4j/graph_paysim.ipynb Imports are incorrectly sorted and/or formatted."  I'm not really clear on what the issue here is.  Any advice?

**Comment by kweinmeister** [comment]
Glad the CLA issue is sorted out!

It looks like `isort` is flagging the `graph_paysim.ipynb` notebook as having import statements that are out-of-order.

Does running `nbqa isort` tool on this notebook provide any useful output? I looked through the notebook, and didn't see anything obvious. Maybe it's getting stuck on something related to import statements being spread throughout the notebook (which shouldn't be a problem).

A workaround if needed could be to bypass an individual import or even the whole file, as described [here](https://github.com/PyCQA/isort#skip-processing-of-imports-outside-of-configuration). Hopefully, we are able to pinpoint the issue, though.

**Comment by MarcoGorelli** [comment]
Yes, it's the extra blank line between the imports in cell 7 - if you run
```console
$ nbqa isort <file name>
```
it'll fix it for you

**Comment by benofben** [comment]
Thank you!

**Comment by benofben** [comment]
Hi @MarcoGorelli, just curious if you had any luck re running this, or if I should try to figure out how to.

Thanks,
Ben

**Comment by kweinmeister** [comment]
@benofben I believe that @MarcoGorelli is the author of the `nbqa` tool, and was just providing some helpful advice. So, I think the next steps here would be to update the notebook accordingly and then update the PR.

**Comment by MarcoGorelli** [comment]
Hey @benofben - if you run
```
nbqa isort notebooks/community/neo4j/graph_paysim.ipynb
```
from the command-line, then `git add` and `git commit` and `git push` again, does it work?

**Comment by benofben** [comment]
I just re ran nbqa.  However, there was no output and it doesn't seem to have modified the file in any way.  There don't seem to be any changes to commit.

**Comment by kweinmeister** [comment]
I was able to reproduce @benofben's result with no `nbqa` output using the latest version, 1.2.2. After downgrading to 0.6.0, the version used by our CI system, I was able to get this output:

```shell
> nbqa isort graph_paysim.ipynb  --nbqa-diff
Cell 7
------
--- graph_paysim.ipynb
+++ graph_paysim.ipynb
@@ -1,3 +1,2 @@
 import pandas as pd
-
 from neo4j import GraphDatabase

Fixing /privategraph_paysim.ipynb
```

Once I removed the line between the import and from, as @MarcoGorelli suggested, the error disappeared on 0.6.0. FYI @MarcoGorelli that the newer version of `nbqa` is not surfacing this message on 1.2.2, even though the `isort` version has not changed.

**Comment by MarcoGorelli** [comment]
Thanks for letting me know! 

The old nbqa version would detect neo4j as a first party package, while the newer version doesn't.

I think the current behaviour is correct - if you ran  Python from the root of the repo, then it wouldn't treat neo4j as a first party import, unless you specified `notebooks` as the root of the repo.

I'd suggest upgrading the CI version, there's been several improvements since 0.6.0

File: notebooks/community/neo4j/graph_paysim.ipynb [hunk 1]
```diff
@@ -0,0 +1,1139 @@
+{
```
Comment (kweinmeister) [inline notebooks/community/neo4j/graph_paysim.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #13.        df = pd.DataFrame(result)
</pre><p>Across this notebook, the dataframe assignment and display occurs both within and outside <code>with</code>  statements. Consider following the pattern of the 2nd query, where it is outside.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (kweinmeister) [inline notebooks/community/neo4j/graph_paysim.ipynb:1]: <p>Typo: unlabled </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (kweinmeister) [inline notebooks/community/neo4j/graph_paysim.ipynb:1]: <p>Is this supposed to be a markdown cell?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (kweinmeister) [inline notebooks/community/neo4j/graph_paysim.ipynb:1]: <p>AutoML Tables is no longer a standalone product in Vertex AI, so I suggest the naming "Vertex AI for AutoML tabular data."</p><p> </p><p>Reference: <a href="https://cloud.google.com/vertex-ai/docs/datasets/prepare-tabular" target="_blank">https://cloud.google.com/vertex-ai/docs/datasets/prepare-tabular</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (kweinmeister) [inline notebooks/community/neo4j/graph_paysim.ipynb:1]: <p>Please consider additional cleanup steps for all assets:</p><p><code></code><code></p><p>endpoint.undeploy_all()</p><p>endpoint.delete()</p><p>model.delete()</p><p>dataset.delete()</p><p></code><code></code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/173/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='72'/>

Comment (benofben) [reply notebooks/community/neo4j/graph_paysim.ipynb:1]: Done

Comment (benofben) [reply notebooks/community/neo4j/graph_paysim.ipynb:1]: Done

Comment (benofben) [reply notebooks/community/neo4j/graph_paysim.ipynb:1]: I think that'd be a good idea.  Would it be ok to get this PR in and then work with @htappen on that?

Comment (kweinmeister) [reply notebooks/community/neo4j/graph_paysim.ipynb:1]: Sure, we can defer this until after the initial PR gets merged.

**Review by kweinmeister — APPROVED** [review]

**Pull request merged**